Feat: Endpoints exposed

TheZackMatthews opened the pull request:
The following endpoints are available now: 
/getDailySurveys, Called when exporting data to .csv
/saveVolunteers,  Called in client when users begin their surveying day
/getAllColunteers, Potentially called in Supervisor app, not strictly necessary
/saveSurvey,  Called from client. Save a new survey and update the corresponding dailySurvey
/getSurveys, Potentially called in Supervisor app? Not strictly necessary

saveSurvey and getSurveys requires req.body.volunteersId, which is provided as a response after a successful post request to /saveVolunteers

Commit:
Feat: getSurveys route

I'm unsure exactly how to work with the one to many relationship between colunteers and survey in controllers, but the models are correct.

Commit:
Fix: Minor bug fix on connection

Commit:
Feat: Endpoints exposed

The following endpoints are available now: getDailySurveys, saveVolunteers, saveSurvey, getSurveys

Commit:
Fix: Working off .env file

Commit:
Fix: Working off .env file

Commit:
Feat: Saving a report should also update dailySurveys

Commit:
Feat: Exposed getAllColunteers endpoint.

Commit:
Fix: Data validation on fish_count on survey model

Commit:
Fix: saveSurvey function now updates the comment array in dailySurveys

This is untested, but feels nessissary

super-bock commented on src/controllers/survey.js:
```diff
@@ -0,0 +1,56 @@
+      volunteersId: id,
+    });
+    // update the value in dailySurveys
+    const isSaved = updateDailySurveys(
```
I am not sure we should be inserting data in two places. We insert in one table and aggregate later. Not really sure what this is for. Let's have a heads up meeting to look at the schema together?

super-bock commented on src/controllers/volunteers.js:
```diff
@@ -0,0 +1,38 @@
+    volunteers.create({
+      creek_name: newVolunteers.creek_name,
+      team_lead: newVolunteers.team_lead,
+      team_members: newVolunteers.team_members,
```
team members should be part of a team table that the volunteer is in

super-bock commented on src/models/dailySurveys.js:
```diff
@@ -0,0 +1,98 @@
+const Sequelize = require('sequelize');
+const { sequelizeConnection, DataTypes } = require('./index.js');
+
+const allSurveys = sequelizeConnection.define('dailySurveys', {
```
Oh ok, so this is for aggregate report or the day? We wouldn't create a table for that in the database, we would create a report by running a query on the database.

super-bock commented on src/models/volunteers.js:
```diff
@@ -0,0 +1,60 @@
+const survey = require('./survey');
+const dailySurveys = require('./dailySurveys');
+
+const newVolunteers = sequelizeConnection.define('newVolunteers', {
```
why is this called 'new'? just volunteers?

TheZackMatthews replied:
We're doing that just because we imported the model as volunteers already, but that naming convention can be changed if it's your preference!

super-bock commented on src/models/volunteers.js:
```diff
@@ -0,0 +1,60 @@
+  ended_at: {
+    type: DataTypes.DATE(),
+  },
+  distance_walked: {
```
this will be fun to calculate from the data points they saved!

super-bock reviewed (commented):
Hey! Let's take a look together when you have time, could be tomorrow early afternoon or Monday. Looks really good, but a couple of things I wasn't sure of.

Commit:
Fix: Removed unneeded code

Strategy changed from keeping a record of each day's data explicitly as a table, to querying that data and formatting it in the supervisor app
